pkg/models/commandvalue.go

```diff
@@ -228,7 +228,7 @@ func NewCommandValue(DeviceResourceName string, origin int64, value interface{},
 	switch t {
 	case Binary:
 		// assign cv.BinValue
-		err = encodeBinaryValue(cv, value)
+		cv.BinValue = value.([]byte)
```

Comment:
If you make this change, the same corresponding change should be made to CommandValues's `BinaryValue()` to remove the call to `decodeBinaryValue()`.

Then the question is where/when is the encode/decode occurring?

Reply:
I thought I discussed where the encode happens in my comment in the associated issue yesterday: https://github.com/edgexfoundry/device-sdk-go/issues/345#issuecomment-544627433

There is some CBOR encoding happening here:

```if event.HasBinaryValue() {
			// TODO: Add conditional toggle in case caller of command does not require this response.
			// Encode response as application/CBOR.
			if len(event.EncodedEvent) <= 0 {
				var err error
				event.EncodedEvent, err = common.EventClient.MarshalEvent(event.Event)
				if err != nil {
					common.LoggingClient.Error("DeviceCommand: Error encoding event", "device", event.Device, "error", err)
				} else {
					common.LoggingClient.Trace("DeviceCommand: EventClient.MarshalEvent encoded event", "device", event.Device, "event", event)
				}
			} else {
				common.LoggingClient.Trace("DeviceCommand: EventClient.MarshalEvent passed through encoded event", "device", event.Device, "event", event)
			}
			// TODO: Resolve why this header is not included in response from Core-Command to originating caller (while the written body is).
			w.Header().Set(clients.ContentType, clients.ContentTypeCBOR)
			w.Write(event.EncodedEvent)
		} else {
			w.Header().Set(clients.ContentType, clients.ContentTypeJSON)
			json.NewEncoder(w).Encode(event)
		}
		// push to Core Data
		go common.SendEvent(event)```

In `restfuncs.go:commandFunc()`

I agree with you that looking at the doc strings for the BinValue field on CommandValue and the `BinaryValue()` method that the intention was to keep *encoded* CBOR inside the BinValue field.

It's very likely I'm missing some context here--but why is the snippet above encoding into CBOR again, when the BinValue is already encoded?  Is the ultimate goal to receive a CBOR encoded event that when decoded contains CBOR encoded data?